srtcore/api.cpp

```diff
@@ -1133,21 +1166,31 @@ SRTSOCKET CUDTUnited::accept(const SRTSOCKET listen, sockaddr* pw_addr, int* pw_
    if (s == NULL)
       throw CUDTException(MJ_SETUP, MN_CLOSED, 0);
 
+   // Set properly the SRTO_GROUPCONNECT flag
+   s->core().m_OPT_GroupConnect = 0;
+
```

Comment:
It's taken care of that it is properly set, that is:
 - it's always 0 on the accepted socket, if the connection is not for a group member
 - then it's checked if this is a group member, if so, it's set to 1

This flag is meaningful in two cases:

1. On the listener socket - the user sets it explicitly to allow groupwise connections to be accepted.
2. Set internally on the accepted socket - this exactly here - so that a user can check this flag in the listener callback.

Reply:
Note of course that this is an old code and this aspect wasn't changed here. Only the locking rules for the group.